File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] freq = new int[2001];
        for (int num : nums) {
            freq[num + 1000]++;
        }
        PriorityQueue<int[]> pq = new PriorityQueue<>(
            (a, b) -> b[0] - a[0]
        );
        for (int i = 0; i < 2001; i++) {
            if (freq[i] > 0) {
                pq.add(new int[]{freq[i], i - 1000});
            }
        }
        
        int[] ans = new int[k];
        for (int i = 0; i < k; i++) {
            ans[i] = pq.poll()[1];
        }
        
        return ans;
    }
}

File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<bool> canReach(n, false);
        canReach[0] = true;
        for (int i = 0; i < n; i++) {
            if (!canReach[i]) continue;
            for (int j = i + 1; j <= min(i + nums[i], n - 1); j++) {
                canReach[j] = true;
            }
        }

        return canReach[n - 1];
    }
};